### PR TITLE
feat(support): generate library

### PR DESCRIPTION
**Pull request by devbww**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11469)

### CODE REVIEW

**Comment by devbww** [comment]
Hmmm.
```
/workspace/cmake-out/google/cloud/support/support_quickstart
google::cloud::Status thrown: UNIMPLEMENTED: Permanent error in ListCases: Method not found.
```

File: google/cloud/support/doc/override-endpoint.dox [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*!
+@page support-override-endpoint How to Override the Default Endpoint
+
+In some cases, you may need to override the default endpoint used by the client
+library. Use the
+[`EndpointOption`](@ref google::cloud::EndpointOption) when initializing the
```
Comment (coryan) [inline google/cloud/support/doc/override-endpoint.dox:6]: Probably my fault, this should be `[\c EndpointOption]` Doxygen gets confused about backticks in links.

File: google/cloud/support/doc/override-authentication.dox [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/*!
+@page support-override-authentication How to Override the Authentication Credentials
+
+Unless otherwise configured, the client libraries use
+[Application Default Credentials] to authenticate with Google Cloud Services.
+While this works for most applications, in some cases you may need to override
+this default. You can do so by providing the
+[`UnifiedCredentialsOption`](@ref google::cloud::UnifiedCredentialsOption)
```
Comment (coryan) [inline google/cloud/support/doc/override-authentication.dox:8]: Another change from backticks to `[\c ...]`.

**Review by coryan — COMMENTED** [review]
I think I introduced a couple of bugs in the scaffold generator.  Easy to fix by hand in this case.

I would ask the service team about that UNIMPLEMENTED error.

**Comment by devbww** [comment]
> Hmmm.
> 
> ```
> /workspace/cmake-out/google/cloud/support/support_quickstart
> google::cloud::Status thrown: UNIMPLEMENTED: Permanent error in ListCases: Method not found.
> ```

Being added to the allow-list seems to have addressed this.

**Review by coryan — APPROVED** [review]

**Pull request merged**